lib/gingr/solr_indexer.rb

```diff
@@ -1,49 +1,45 @@
 # frozen_string_literal: true
 require 'faraday/net_http_persistent'
+require 'find'
 require 'rsolr'
 require_relative 'config'
 
 module Gingr
   class SolrIndexer
     include Config
 
-    attr_reader :reference_urls
-    attr_reader :solr
+    attr_accessor :reference_urls
+    attr_accessor :solr
 
-    def initialize(url, reference_urls = {})
-      @solr = RSolr.connect url:, adapter: :net_http_persistent
-      @reference_urls = reference_urls
+    def initialize(solr = nil, reference_urls = nil)
+      solr ||= ENV['SOLR_URL'] || Gingr::Config.default_options[:solr_url]
+      solr = RSolr.connect url: solr, adapter: :net_http_persistent if solr.kind_of? String
+      @solr = solr
+      @reference_urls = reference_urls || {}
     end
 
-    def update_reference_urls?
-      !@reference_urls.empty?
+    def add(doc)
+      doc = JSON.load_file(doc) if doc.kind_of? String
```

Comment:
This is similar to the initialization logic above. If you want to pass a parsed document (hash) you can, but for convenience you can also pass a filepath, which is parsed as JSON.

Reply:
It's more flexible now